Setup/MakePortableArchive.cmd

```diff
@@ -8,281 +8,778 @@ cd /d "%~p0"
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Security.Cryptography.Csp.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Security.Cryptography.Encoding.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Security.Cryptography.Primitives.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Security.Cryptography.X509Certificates.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Security.Principal.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Security.SecureString.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Text.Encoding.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Text.Encoding.Extensions.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Text.RegularExpressions.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Threading.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Threading.Overlapped.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Threading.Tasks.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Threading.Tasks.Parallel.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Threading.Thread.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Threading.ThreadPool.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Threading.Timer.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Xml.ReaderWriter.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Xml.XDocument.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Xml.XmlDocument.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Xml.XmlSerializer.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Xml.XPath.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1 
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\System.Xml.XPath.XDocument.dll GitExtensions\
+IF ERRORLEVEL 1 EXIT /B 1
+
+REM Resources from dependencies
+echo "Resource libs"
+xcopy /y /i ..\GitExtensions\bin\%Configuration%\cs\Microsoft.VisualStudio.Threading.resources.dll GitExtensions\cs\
```

Comment:
We have been explicit in this script.  I wrote a script to parse the script for files.  It then looks in gitextensions\bin\release folder and gathers all files hashing all.  I gathered all the shortest path files by hash and generated new xcopy lines.  I ignored some that are handled by folder copy and ones that should not be included.  

I am working on a release script manager tool.  It will allow us to manage these scripts by running a utility.

Reply:
I put in sections to help.

Reply:
The benefit of this way is the ci build will tell us exactly where the missing file is.  It will try to xcopy a file that doesn't exist and fail and exit.  The output will show you the line that it just tried to copy.